doc/src/manual/parallel-computing.md

```diff
@@ -1,12 +1,542 @@
 # Parallel Computing
 
-This part of the manual details the following types of parallel programming available in Julia.
+For any newcomer to multi-thread and parallel computing it can be useful to first appreciate 
+the different levels of parallelism offered by Julia. We can divide them in three main categories :
 
-1. Distributed memory using multiple processes on one or more nodes
-2. Shared memory using multiple processes on a single node
-3. Multi-threading
+1. Julia Coroutines (Green Threading)
+2. Multi-Threading
+3. Multi-Core or Distributed Processing
 
-# Distributed Memory Parallelism
+We will consider first Julia [Tasks (aka Coroutines)](@ref man-tasks) and other high-level interfaces and
+modules that rely on the Julia runtime library, that allow to suspend and resume computations with full
+control of inter-`Tasks` communication without worrying of the operative system's scheduler. Julia also
+allows to communicate between `Tasks` through operations like [`wait`](@ref) and [`fetch`](@ref).
+Communication and data synchronization is managed through [`Channel`](@ref)s, which are the conduit
+that allows inter-`Tasks` communication.
+
+Julia also supports multi-threading, where execution is forked and an anonymous function is run across all
```

Comment:
I would reformulate this a bit.

Julia also supports experimental multi-threading, through a fork-join approach.

Then you quite accurately describe what fork-join means.

Reply:
Do you think that the description that I added now is too short?